homeassistant/components/geo_json_events/config_flow.py

```diff
@@ -0,0 +1,130 @@
+        if legacy_scan_interval and isinstance(legacy_scan_interval, timedelta):
+            import_config[CONF_SCAN_INTERVAL] = legacy_scan_interval.total_seconds()
+        url: str = import_config[CONF_URL]
+        latitude: float | None = import_config.get(
+            CONF_LATITUDE, self.hass.config.latitude
+        )
+        longitude: float | None = import_config.get(
+            CONF_LONGITUDE, self.hass.config.longitude
+        )
+        self._async_abort_entries_match(
+            {
+                CONF_URL: url,
+                CONF_LATITUDE: latitude,
+                CONF_LONGITUDE: longitude,
+            }
+        )
+        return self.async_create_entry(
+            title=f"{url} ({latitude}, {longitude})",
+            data={
+                CONF_URL: url,
+                CONF_LATITUDE: latitude,
+                CONF_LONGITUDE: longitude,
+                CONF_RADIUS: import_config.get(CONF_RADIUS, DEFAULT_RADIUS_IN_KM),
+                CONF_SCAN_INTERVAL: import_config.get(
+                    CONF_SCAN_INTERVAL, DEFAULT_SCAN_INTERVAL
+                ),
+            },
+        )
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the start of the config flow."""
+        if not user_input:
+            suggested_values: Mapping[str, Any] = {
+                CONF_LOCATION: {
+                    CONF_LATITUDE: self.hass.config.latitude,
+                    CONF_LONGITUDE: self.hass.config.longitude,
+                    CONF_RADIUS: DEFAULT_RADIUS_IN_M,
+                }
+            }
+            data_schema = self.add_suggested_values_to_schema(
+                DATA_SCHEMA, suggested_values
+            )
+            return self.async_show_form(
+                step_id="user",
+                data_schema=data_schema,
+            )
+
+        url: str = user_input[CONF_URL]
+        latitude: float | None = user_input.get(CONF_LOCATION, {}).get(CONF_LATITUDE)
+        longitude: float | None = user_input.get(CONF_LOCATION, {}).get(CONF_LONGITUDE)
+        self._async_abort_entries_match(
+            {
+                CONF_URL: url,
+                CONF_LATITUDE: latitude,
+                CONF_LONGITUDE: longitude,
+            }
+        )
+        return self.async_create_entry(
+            title=f"{url} ({latitude}, {longitude})",
+            data={
+                CONF_URL: url,
+                CONF_LATITUDE: latitude,
+                CONF_LONGITUDE: longitude,
+                CONF_RADIUS: DistanceConverter.convert(
+                    user_input.get(CONF_LOCATION, {}).get(
+                        CONF_RADIUS, DEFAULT_RADIUS_IN_M
+                    ),
+                    UnitOfLength.METERS,
+                    UnitOfLength.KILOMETERS,
+                ),
+                CONF_SCAN_INTERVAL: user_input.get(
+                    CONF_SCAN_INTERVAL, DEFAULT_SCAN_INTERVAL
+                ),
```

Comment:
It is marked as required in the schema.
```suggestion
                CONF_SCAN_INTERVAL: user_input[CONF_SCAN_INTERVAL],
```

Reply:
Yes, makes sense.